apps/smart-contracts/core/test/RedeemHook.test.ts

```diff
@@ -0,0 +1,105 @@
+import chai, { expect } from 'chai'
+import { ethers } from 'hardhat'
+import { SignerWithAddress } from '@nomiclabs/hardhat-ethers/dist/src/signer-with-address'
+import { id, parseEther } from 'ethers/lib/utils'
+import { Contract } from 'ethers'
+import { MockContract, FakeContract, smock } from '@defi-wonderland/smock'
+import { ZERO_ADDRESS } from 'prepo-constants'
+import { redeemHookFixture, smockAccountListFixture } from './fixtures/HookFixture'
+import { fakeTokenSenderFixture } from './fixtures/TokenSenderFixture'
+import { smockTestERC20Fixture } from './fixtures/TestERC20Fixture'
+import { RedeemHook } from '../typechain'
+
+chai.use(smock.matchers)
+
+describe('=> RedeemHook', () => {
+  let deployer: SignerWithAddress
+  let user: SignerWithAddress
+  let treasury: SignerWithAddress
+  let redeemHook: RedeemHook
+  let allowlist: MockContract<Contract>
+  let baseToken: MockContract<Contract>
+  let tokenSender: FakeContract<Contract>
+
+  beforeEach(async () => {
+    ;[deployer, user, treasury] = await ethers.getSigners()
+    redeemHook = await redeemHookFixture()
+    allowlist = await smockAccountListFixture()
+    baseToken = await smockTestERC20Fixture('Test Token', 'TEST', 18)
+    tokenSender = await fakeTokenSenderFixture(baseToken.address)
+  })
+
+  describe('initial state', () => {
+    it('sets deployer to owner', async () => {
+      expect(await redeemHook.owner()).to.eq(deployer.address)
+    })
+
+    it('sets nominee to zero address', async () => {
+      expect(await redeemHook.getNominee()).to.eq(ZERO_ADDRESS)
+    })
+  })
+
+  describe('setAllowlist', () => {
```

Comment:
nit: (same for others)
```suggestion
  describe('# setAllowlist', () => {
```